soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -1796,53 +1824,21 @@ std::string AltarIconString(char iconChar) {
 
 std::string FormatJsonHintText(std::string jsonHint) {
     std::string formattedHintMessage = jsonHint;
-    char newLine = 0x01;
-    char playerName = 0x0F;
-    char nextBox = 0x04;
-    std::replace(formattedHintMessage.begin(), formattedHintMessage.end(), '&', newLine);
-    std::replace(formattedHintMessage.begin(), formattedHintMessage.end(), '^', nextBox);
-    std::replace(formattedHintMessage.begin(), formattedHintMessage.end(), '@', playerName);
 
     std::unordered_map<std::string, char> textBoxSpecialCharacters = {
-        {"À", 0x80 },
-        {"î", 0x81 },
-        {"Â", 0x82 },
-        {"Ä", 0x83 },
-        {"Ç", 0x84 },
-        {"È", 0x85 },
-        {"É", 0x86 },
-        {"Ê", 0x87 },
-        {"Ë", 0x88 },
-        {"Ï", 0x89 },
-        {"Ô", 0x8A },
-        {"Ö", 0x8B },
-        {"Ù", 0x8C },
-        {"Û", 0x8D },
-        {"Ü", 0x8E },
-        {"ß", 0x8F },
-        {"à", 0x90 },
-        {"á", 0x91 },
-        {"â", 0x92 },
-        {"ä", 0x93 },
-        {"ç", 0x94 },
-        {"è", 0x95 },
-        {"é", 0x96 },
-        {"ê", 0x97 },
-        {"ë", 0x98 },
-        {"ï", 0x99 },
-        {"ô", 0x9A },
-        {"ö", 0x9B },
-        {"ù", 0x9C },
-        {"û", 0x9D },
-        {"ü", 0x9E }
+        { "À", 0x80 }, { "î", 0x81 }, { "Â", 0x82 }, { "Ä", 0x83 }, { "Ç", 0x84 }, { "È", 0x85 }, { "É", 0x86 },
+        { "Ê", 0x87 }, { "Ë", 0x88 }, { "Ï", 0x89 }, { "Ô", 0x8A }, { "Ö", 0x8B }, { "Ù", 0x8C }, { "Û", 0x8D },
+        { "Ü", 0x8E }, { "ß", 0x8F }, { "à", 0x90 }, { "á", 0x91 }, { "â", 0x92 }, { "ä", 0x93 }, { "ç", 0x94 },
+        { "è", 0x95 }, { "é", 0x96 }, { "ê", 0x97 }, { "ë", 0x98 }, { "ï", 0x99 }, { "ô", 0x9A }, { "ö", 0x9B },
+        { "ù", 0x9C }, { "û", 0x9D }, { "ü", 0x9E }
```

Comment:
it seems with your changes this stuff exists in custommessagemanager now too? any ideas on how to reduce copypasta here?

Reply:
It exists here because this is for pulling the text from the spoilerfile into the saveContext (which then gets pulled into the custom message tables afterwards). I could expose `FormatMessage` instead of it being a private method though.

Reply:
i don't think making `FormatMessage` public really hurts anything, but if you run into errors and it ends up taking more than like 10 minutes to switch over to using it then removing this copypasta can be considered out of scope for this pr